sync/clients/sync.py

```diff
@@ -13,78 +14,110 @@
+
+    def _set_cached_token(self) -> None:
+        raise NotImplementedError
+
+    def _get_cached_token(self) -> Optional[Tuple[str, datetime]]:
+        raise NotImplementedError
+
+
+class FileCachedToken(CachedToken):
+    def __init__(self):
+        self._cache_file = Path(user_cache_dir("syncsparkpy")) / "auth.json"
+
+        super().__init__()
 
     def _get_cached_token(self) -> Optional[Tuple[str, datetime]]:
         # Cache is optional, we can fail to read it and not worry
-        token_file = self._cache_file / "access_token"
-        if token_file.exists():
+        if self._cache_file.exists():
             try:
-                cached_token = json.loads(token_file.read_text())
-                cached_expiry = datetime.fromisoformat(cached_token["expires_at_utc"])
+                cached_token = json.loads(self._cache_file.read_text())
                 cached_access_token = cached_token["access_token"]
-                return (cached_access_token, cached_expiry)
+                cached_expiry = datetime.fromisoformat(cached_token["expires_at_utc"])
+                return cached_access_token, cached_expiry
             except Exception as e:
                 logger.warning(
-                    f"Failed to read cached access token @ {token_file}", exc_info=e
+                    f"Failed to read cached access token @ {self._cache_file}", exc_info=e
                 )
+
         return None
 
-    def _set_cached_token(self, access_token: str, expires_at_utc: datetime):
+    def _set_cached_token(self) -> None:
         # Cache is optional, we can fail to read it and not worry
-        token_file = self._cache_file / "access_token"
         try:
-            token_file.write_text(
+            self._cache_file.parent.mkdir(parents=True, exist_ok=True)
+            self._cache_file.write_text(
                 json.dumps(
                     {
-                        "access_token": access_token,
-                        "expires_at_utc": expires_at_utc.isoformat(),
+                        "access_token": self._access_token,
+                        "expires_at_utc": self._access_token_expires_at_utc.isoformat(),
                     }
                 )
             )
         except Exception as e:
             logger.warning(
-                f"Failed to write cached access token @ {token_file}", exc_info=e
+                f"Failed to write cached access token @ {self._cache_file}", exc_info=e
             )
 
-    @property
-    def _access_token_valid(self) -> bool:
-        if not self._access_token:
-            return False
-        if self._access_token_expires_at_utc:
-            return datetime.now(
-                tz=timezone.utc
-            ) < self._access_token_expires_at_utc - timedelta(seconds=20)
-        return False
+
+class SyncAuth(httpx.Auth):
+    requires_response_body = True
+
+    def __init__(self, api_url: str, api_key: APIKey, cached_token_cls: Type[CachedToken] = FileCachedToken):
```

Comment:
The plan here is to subclass `CachedToken` from `sync-svc-library-tasks` to support redis caching, the file cache behavior should match what was there before.